src/database/database-leveldb.cpp

```diff
@@ -97,5 +98,100 @@ void Database_LevelDB::listAllLoadableBlocks(std::vector<v3s16> &dst)
 	delete it;
 }
 
-#endif // USE_LEVELDB
+AuthDatabaseLevelDB::AuthDatabaseLevelDB(const std::string &savedir)
+{
+	leveldb::Options options;
+	options.create_if_missing = true;
+	leveldb::Status status = leveldb::DB::Open(options,
+		savedir + DIR_DELIM + "auth.db", &m_database);
+	ENSURE_STATUS_OK(status);
+}
+
+AuthDatabaseLevelDB::~AuthDatabaseLevelDB()
+{
+	delete m_database;
+}
+
+bool AuthDatabaseLevelDB::getAuth(const std::string &name, AuthEntry &res)
+{
+	std::string raw;
+	leveldb::Status s = m_database->Get(leveldb::ReadOptions(), name, &raw);
+	if (!s.ok())
+		return false;
+	std::istringstream is(raw);
+
+	/*
+	u8 version = 0
+	std::string password
+	u16 number of privileges
+	for each privilege {
+		std::string privilege
+	}
+	s64 last_login
+	*/
+
+	if (readU8(is) != 0)
```

Comment:
For future changes, following code would help to reduce missing adjustments:
`if (readU8(is) > AUTH_VERSION)` where `AUTH_VERSION` is a macro defined above (and undefined later).